client/packages/coldchain/src/Equipment/ImportAsset/UploadTab.tsx

```diff
@@ -0,0 +1,173 @@
+  onUploadComplete: () => void;
+  catalogueItemData?: AssetCatalogueItemFragment[];
+}
+
+export const EquipmentUploadTab: FC<ImportPanel & EquipmentUploadTabProps> = ({
+  tab,
+  setErrorMessage,
+  setEquipment,
+  onUploadComplete,
+  catalogueItemData,
+}) => {
+  const t = useTranslation('coldchain');
+  const { error } = useNotification();
+  const [isLoading, setIsLoading] = useState(false);
+  const EquipmentBuffer: EquipmentImportModal.ImportRow[] = [];
+
+  const csvExample = async () => {
+    const emptyRows: ImportRow[] = [];
+    const csv = importEquipmentToCsv(
+      emptyRows.map((row: ImportRow): InsertAssetInput => {
+        return toInsertEquipmentInput(row, catalogueItemData);
+      }),
+      t
+    );
+    FileUtils.exportCSV(csv, t('filename.cce'));
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const csvImport = (files: any) => {
+    setErrorMessage('');
+    EquipmentBuffer.length = 0; // Reset the import buffer
+    const csvFile = files[0];
+
+    if (!csvFile.name.endsWith('.csv')) {
+      setErrorMessage(t('messages.invalid-file'));
+      return;
+    }
+
+    if (csvFile) {
+      setIsLoading(true);
+      Papa.parse(csvFile, {
+        header: true,
+        worker: true,
+        skipEmptyLines: true,
+        chunkSize: 100 * 1024, // 100kb
+        chunk: processUploadedDataChunk,
+        complete: () => {
+          setEquipment(EquipmentBuffer);
+          setIsLoading(false);
+          onUploadComplete();
+        },
+      });
+    }
+    error(t('messages.error-no-file-selected'));
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const processUploadedDataChunk = (data: any) => {
+    if (!data.data || !Array.isArray(data.data)) {
+      // Don't think this is likely to happen...
+      console.info('data not data');
+      setErrorMessage(t('messages.import-error'));
+    }
+
+    const rows: EquipmentImportModal.ImportRow[] = [];
+    let hasErrors = false;
+    for (const row of data.data) {
+      const importRow = {} as EquipmentImportModal.ImportRow;
+      const rowErrors: string[] = [];
+      if (row.id && row.id.trim() != '') {
+        importRow.id = row.id;
+        importRow.isUpdate = true;
+      } else {
+        importRow.id = FnUtils.generateUUID();
+        importRow.isUpdate = false;
+      }
+      if (
+        row[t('label.asset-number')] &&
+        row[t('label.asset-number')].trim() != ''
+      ) {
+        importRow.assetNumber = row[t('label.asset-number')];
+      } else {
+        rowErrors.push(
+          t('error.field-must-be-specified', { field: t('label.asset-number') })
+        );
+      }
+      if (
+        row[t('label.catalogue-item-code')] === undefined ||
+        row[t('label.catalogue-item-code')].trim() === ''
+      ) {
+        rowErrors.push(
+          t('error.field-must-be-specified', {
+            field: t('label.catalogue-item-code'),
+          })
+        );
+      } else if (
```

Comment:
Additional check that catalogue item code matches the code of the upload sheet.

Maybe this is a helpful data validation or maybe not? It is possible to create assets without a catalogue item ID, so can remove this. However maybe it is good to know if a user thinks they have a code but the code doesn't match a catalogue item?

Current behaviour is that the upload cannot go ahead if a single code doesn't match a catalogue item

Reply:
you can create without a catalogue item through the UI.. but let's leave it off the import for now

Reply:
Do you think best to not include code at all in the import csv?

Reply:
no - keep the code, I think it's good to have.